### PR TITLE
Add plain JavaScript support with `blitz new myapp --js`

### PR DESCRIPTION
**Pull request by aem**
### Type: Feature 🆕
Closes: #160 

### What are the changes and their implications? :gear:

We now fully support JavaScript file generation out of the box! If the user supplies the `--js` flag to `blitz new` we'll convert all file contents to JS using Babel (and move all file extensions to be `.js`) before committing our changes, and we'll ignore generation of the `tsconfig.json` for these projects as well. In future instances when calling `blitz generate`, we'll look for the `tsconfig.json` and only generate TS files for projects that already have a `tsconfig.json`.

Making this automatic based on the config means that if you'd like to migrate to TS at a later date it's as simple as adding a `tsconfig.json` and we'll start generating your files in TS instead of JS. Currently there's no way to go in the other direction (generate JS files in a TS project), but if we find a significant need we can figure that out.

~This depends on the changes in #281 so currently this branch is based off of that one, I'll rebase against `canary` once that gets merged.~

### Checklist

- [ ] Tests added for changes
- [ ] Any added terminal logging uses `packages/server/src/log.ts`

### CODE REVIEW

**Comment by aem** [comment]
> blitz generate still generates TS files inside a JS project

Oof. Looks like I accidentally dropped a commit while rebasing against `canary` 😨 Let me try to revive that

File: packages/cli/src/commands/new.ts [hunk 1]
```diff
@@ -26,10 +26,9 @@ export default class New extends Command {
 
   static flags = {
     help: flags.help({char: 'h'}),
-    ts: flags.boolean({
-      description: 'generate a TypeScript project',
-      default: true,
-      allowNo: true,
+    js: flags.boolean({
+      description: 'Generates a JS project. Omitting this flag will TypeScript as the default language.',
```
Comment (flybayer) [inline packages/cli/src/commands/new.ts:30]:
```suggestion
      description: 'Generates a JS project. TypeScript is the default unless you add this flag.',
```

I went to fix a typo and decided to tweak it a bit

**Comment by flybayer** [comment]
Still generating TS

1. `yarn link-cli` in root
2. Outside monorepo somewhere, run `blitz new -h` to verify the flag is there
3. `blitz new myapp --js`
4. JS files are generated properly here
5. `cd myapp`
6. `blitz g all user`
7. Fail: ts files are generated

**Comment by aem** [comment]
if you run `blitz new -h` in between 5 and 6 there what do you get @flybayer? I think once you install deps in the new project it's going to use the locally installed versinon of `blitz` rather than the linked/global one

**Comment by flybayer** [comment]
Oh yes 🤦

**Pull request merged**